configs/common/packages.yaml

```diff
@@ -9,290 +9,255 @@
         glu: [openglu]
         jpeg: [libjpeg-turbo]
         lapack: [openblas]
+        pkgconfig: [pkg-config]
         yacc: [bison]
         zlib-api: [zlib-ng]
     #
     awscli-v2:
-      require: "~examples"
+      require: '~examples'
     bacio:
-      version: ['2.4.1']
+      require: '@2.4.1'
     bison:
-      version: ['3.8.2']
-    # 1.85 incompatible with ecflow@5.11.4 - use latest "good version"
+      require: '@3.8.2'
+    # 1.85 incompatible with ecflow@5.11.4 - use latest 'good version'
     boost:
-      require: "@1.84 ~atomic +chrono +date_time +exception +filesystem ~graph ~iostreams ~locale ~log ~math ~mpi ~numpy +pic +program_options +python ~random +regex +serialization ~signals +system +test +thread +timer ~wave cxxstd=17 visibility=hidden"
+      require: '@1.84 ~atomic +chrono +date_time +exception +filesystem ~graph ~iostreams ~locale ~log ~math ~mpi ~numpy +pic +program_options +python ~random +regex +serialization ~signals +system +test +thread +timer ~wave cxxstd=17 visibility=hidden'
     bufr:
-      version: ['12.0.1']
-      variants: +python
+      require: '@12.0.1 +python'
     cairo:
-      variants: +pic
+      require: '+pic'
     cdo:
-      version: ['2.2.0']
-      variants: ~openmp
+      require: '@2.2.0 ~openmp'
     cmake:
       version: ['3.27.9']
-      variants: +ownlibs
+      require: '+ownlibs'
     # Attention - when updating also check the various jcsda-emc-bundles env packages
     crtm:
-      version: ['2.4.0.1']
-      variants: +fix
+      require: '+fix'
     ecbuild:
-      version: ['3.7.2']
+      require: '@3.7.2'
     eccodes:
-      version: ['2.33.0']
-      variants: +png
+      require: '@2.33.0 +png'
     ecflow:
-      version: ['5.11.4']
-      variants: +ui
+      require: '@5.11.4 +ui'
     eckit:
-      version: ['1.24.5']
-      variants: linalg=eigen,lapack compression=lz4,bzip2
+      require: '@1.24.5 linalg=eigen,lapack compression=lz4,bzip2'
     ecmwf-atlas:
-      version: ['0.36.0']
-      variants: +fckit +trans +tesselation +fftw
+      require: '@0.36.0 +fckit +trans +tesselation +fftw'
     ectrans:
-      version: ['1.2.0']
-      variants: ~mkl +fftw
+      require: '@1.2.0 ~mkl +fftw'
     eigen:
-      version: ['3.4.0']
+      require: '@3.4.0'
     # Attention - when updating the version also check the common modules.yaml
     # config and update the projections for lmod/tcl.
     # Also, check the acorn and derecho site configs which have esmf modifications.
     esmf:
       require:
-        - any_of: ['@=8.6.1 ~xerces ~pnetcdf snapshot=none +shared +external-parallelio fflags="-fp-model precise" cxxflags="-fp-model precise"']
+        - '@=8.6.1 ~xerces ~pnetcdf snapshot=none +shared +external-parallelio'
```

Comment:
nice!

Reply:
Huh. This might be one of the reasons spack-stack GEOS is non-zero-diff. Never noticed that...